.github/workflows/build_and_test_tidesdb.yml

```diff
@@ -0,0 +1,34 @@
+name: C/C++ CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout tidesdb repo
+      uses: actions/checkout@v4
+
+    - name: Install libzstd-dev,liblz4-dev and libsnappy-dev
+      run: |
+        sudo apt update
+        sudo apt install -y libzstd-dev liblz4-dev libsnappy-dev
+
+    - name: configure cmake build for tidesdb
+      run: | 
+        cd tidesdb
+        cmake --debug-output  -S . -B build && make -C build/
+
+    - name: run tests
+      run: |
+        export ASAN_OPTIONS=verify_asan_link_order=0
+        cd tidesdb/build
+        ctest
```

Comment:
@guycipher this is test command that we run. actually better to run it with `--output-on-failure` flag but anyway. 

so if you want some env variable sets this is good place to set them (via export)

Reply:
it is also possible to set several combinations of env. variables and during build config (line #27) e.t.c.

Reply:
Understood.  I see how it works.  I’m going to office now but have you been able to test this or we have to merge first?  I will assist in testing.

Reply:
Well it is partially used in tidesdb-lua action  and that already been tested well. So I doubt it won't work. And yes we have to merge it first for GH in order to test it unfortinately.

Reply:
You always may just delete the file in the following revert commit if you don't like it - it is not a big deal

Reply:
No problemo.  We shall test it!